<launch>
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <arg name="model" default="$(find vis_lecture)/urdf/robot1_simulation.urdf" />
    <arg name="gui" default="true" />
    <arg name="rvizconfig"  default="$(find vis_lecture)/rviz/robot1_simulation.rviz" />

    <!-- <param name="robot_description" textfile="$(arg model)" /> -->
    <param name="robot_description" command="$(find xacro)/xacro $(arg model)"/>
    <param name="use_gui" value="$(arg gui)" />

    <node name="robot1_simulation_joint_publisher" pkg="vis_lecture" type="robot1_simulation_joint_publisher" output="screen" />
    <!-- <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" /> -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />
    <!-- <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-file $(arg model) -urdf -model robot1_simulation" /> -->
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-param robot_description -urdf -model robot1_simulation" />
</launch>